### ISSUE TITLE
Pass down custom graphql query via Edit, Show, Create Component

### ISSUE THREAD

**Issue by DVGY** [marmelab/react-admin#7357]
React admin is a great library. We have built a large app out of it

**Is your feature request related to a problem? Please describe.**

We are changing our app from REST to GraphQL. We have lots of custom views, and **approx. 100 REST API resources.** A view (Front end UI) maybe shown differently to each person depending on some logic.

While moving from REST to GraphQL, we found two problems, 
1. We did not find any props to pass custom GraphQL Query or Mutation down to the DataProvider for Edit, Create and Show Component. 
2. In the [ra-simple-graphql](https://github.com/marmelab/react-admin/tree/master/packages/ra-data-graphql-simple#overriding-a-specific-query) it is advised to use resource and type of DataProvider to handle custom queries. **If we use this approach then we may need to write Number of resources x Number of DataProvider Methods,  If/else Conditions.
 (100 * 6 = 600) in my case.**

There for we were hoping that we could call custom GraphQL Query which will be passed down via Edit or Show Component. Then we may need not write to 600 Conditions, It will be much less i.e only 6 conditions or 10 conditions.
<img width="917" alt="image" src="https://user-images.githubusercontent.com/46158826/157081370-e04de358-3f9f-460a-af7e-7037710167b4.png">

For example

```

const customGQLQuery = gql`

query User{
  User {
   id,
   address
   }

`

<Edit customGQLQuery = {customGQLQuery}>
<Edit />


const getOne = async (resource,params) => {

if(params.customGQLQuery){
  // execute custom gql Query

}
else {
 // use default query builder

}

```

Since we could not find any ways to pass props down to DataProvider, we have to write lot of code otherwise graphql will call it's default query and bring fields we may not need for the UI
 

**Describe the solution you'd like**

Ability to pass custom prop via for custom GraphQL Query<Edit/>, <Create/> and <Show/>

**Describe alternatives you've considered**
For List component I have used filters to pass custom graphql query.

**Additional context**

It is also possible that I have missed something and my problems solution is already available. In that case pls point out to me specific resource

**Comment by fzaninotto** [comment]
React-admin v4 introduces [the meta parameter](https://marmelab.com/react-admin/doc/4.0/Actions.html#meta-parameter) on all dataProvider queries, so what you ask is already possible by using the `dataProvider` hooks (e.g. `useGetOne`) manually. 

We'll add support for passing a custom `meta` parameter to view components (e.g. `<Edit>`) in a future minor release.

**Comment by DVGY** [comment]
@fzaninotto Thank you so much. I will be waiting eagerly for this.

**Comment by panfiva** [comment]
@[fzaninotto](https://github.com/fzaninotto), do you have estimated timeline for including support for passing custom meta to view components? Thanks

**Comment by fzaninotto** [comment]
Fixed by #7841
